### ISSUE TITLE
Java 9 compatibility broken for QuaquaTextFieldFocusHandler

### ISSUE THREAD

**Issue by tiberiughioca** [karlvr/Quaqua#11]
Are there plans for fixing the compatibility with Java 9 for class QuaquaTextFieldFocusHandler?

**Comment by karlvr** [comment]
Not currently as I wasn’t aware of the issue! Could you please provide more details?

If you’d like to provide a fix it would be welcome.

**Comment by tiberiughioca** [comment]
I am not sure how this could be solved without breaking the compatibility with earlier versions than Java 9...

Here is the actual code from master that is compatible with Java 8:
```
if (event instanceof CausedFocusEvent) {
 CausedFocusEvent cfEvent = (CausedFocusEvent) event;
                    if (cfEvent.getCause() == Cause.TRAVERSAL_FORWARD ||
                            cfEvent.getCause() == Cause.TRAVERSAL_BACKWARD) {
                        tc.selectAll();
                    }
                }
```

And here is the code that is compatible with Java 9 but breaks Java 8 compatibility:
```
if (event.getCause() == Cause.TRAVERSAL_FORWARD ||
                            event.getCause() == Cause.TRAVERSAL_BACKWARD) {
                        tc.selectAll();
                    }
```
The problem is that in Java 9 the class sun.awt.CausedFocusEvent does not exist anymore (it was removed) while in Java 8 the method java.awt.event.FocusEvent.getCause() is not available since it was added in Java 9. Any idea how we could solve this problem?

**Comment by karlvr** [comment]
We can test for the existence of `sun.awt.CausedFocusEvent` and set a variable to switch us between these two implementations?

**Comment by tiberiughioca** [comment]
I do not think this is possible because you get an error even when you just import that class...

The right solution I think it would be to generate a multi-release jar with both implementations. But that would mean more changes regarding the structure of the project and the ant scripts. Details at http://openjdk.java.net/jeps/238

**Comment by karlvr** [comment]
Is the error that you get when you import the class about it being restricted?

**Comment by tiberiughioca** [comment]
This is the error on import: "The type sun.awt.CausedFocusEvent is not visible".
And this is the error when it is used: "CausedFocusEvent cannot be resolved to a type".

And the same goes for class sun.awt.CausedFocusEvent.Cause... It is pretty easy to see these errors if you take the latest code from master and use it with Java 9.

**Comment by karlvr** [comment]
Could the code that uses CausedFocusEvent be changed to work purely with reflection? In that way it wouldn't need to import the restricted class and could avoid this error.

**Comment by brad-eric** [comment]
I was trying Quaqua 9.1 with Java 9 today & i've got the same issue!
Hope this'd be fixed some day.
